src/api/CMakeLists.txt

```diff
@@ -26,7 +26,8 @@ add_library(migraphx_c
     api.cpp
 )
 set_target_properties(migraphx_c PROPERTIES EXPORT_NAME c)
-rocm_set_soversion(migraphx_c 3.0)
+
+rocm_set_soversion(migraphx_c 3.0) # why does it this not get same .so number as other libs ? 
```

Comment:
This is a stable API interface so the version is only bumped when we break binary compatibility.

The other components do not have binary compatibility across versions so we bump the so version along with the migraphx version. Although, we actually do this wrong though. The SO version for 2.6 should be 20600.0 instead of 2.6, because the major version should be bumped on every version. A major version jump says the two SO files are not compatible, whereas a minor version bump means its backwards compatible. So we need to always bump the major version.

Reply:
If we do this wrong, should we fix it right now then instead of building ontop of it?

Reply:
Other  libraries are following 
`
librocm-core.so.1.0.50402
`
kind of versioning. 

Why do they need `1.0` as prefix ?

Reply:
Should be versioning like `2.6.50402` in that case ?

Reply:
>  because the major version should be bumped on every version.

Every version you mean going from rocm-5.x.y to rocm-6.x.y ? Or from going rocm-5.4.2 to rocm-5.5.x ?